reana/cli.py

```diff
@@ -476,9 +478,60 @@ def select_components(components):
         else:
             display_message('Ignoring unknown component {0}.'.format(
                 component))
+
+    if exclude_components:
+        output = exclude_components_from_selection(output, exclude_components)
+
     return list(output)
 
 
+def exclude_components_from_selection(selection, exclude_components):
+    """Exclude list of components from list of selections.
+
+    :param selection: List of selected components in standard naming form.
+    :param exclude_components: List of components to exclude, either in short
+        or standard naming form.
+    :type selection: set
+    :type exclude_components: list
+
+    :return: Set of selected components without ``exclude_components``, all in
+        standard naming form.
+    :rtype: set
+    """
+    def _normalise_component_names(components):
+        """Normalise component name list to standard form.
+
+        It is not possible to widely use ``find_standard_component_name``
+        in the scenario of mixed short and standard names,
+        for example with ``components = ['r-ui', 'reana-server']``
+        an exception will be thrown by ``find_standard_component_name``.
+
+        Instead, we shorten all component names (idempotent operation)
+        and then find the standard component name.
+
+        :param components: List of components
+        :type components: list
+
+        :return: List with all components in standard (long) naming form.
+        :rtype: List
+        """
+        short_components = \
```

Comment:
Hmm, perhaps we could take this normalising function outside, as it might be useful elsewhere?

Or perhaps we should simply enrich `find_standard_component_name()` so that it would accept any input, even mixed? (It was originally written as short name -> standard name converter.)

P.S. There are test cases, we can enrich them as well, to make sure about the exclusion behaviour.

Wondering because the comment *"It is not possible..."* seems like a wrong reason... Something we can easily solve outside the function and it would simplify the logic here...

Reply:
I was not sure we wanted the behaviour to be changed there, much better by making `find_standard_component_name` idempotent, also tests added, sorry about that 😊, it should be now ready.